cmake/options.cmake

```diff
@@ -1,4 +1,5 @@
 include(leatherman)
+defoption(ENABLE_CXX_WERROR "Enables the -Werror compiler option" OFF)
```

Comment:
Don't know if this is the best name for this option.

Reply:
Name seems fine. We're defaulting to off? I'm not keen on ignoring warnings. In the past we've added exceptions where they made sense, or used platform-specific CMAKE_CXX_FLAGS config.

Reply:
Yeah that's something I wasn't sure about. The reason I defaulted to `OFF` here is to avoid having to set it to `OFF` in all of our components that compile with GCC 8. I'm happy to change it to `ON` if that is preferred. @branan Do you have any thoughts?

Reply:
@MikaelSmith now that we're building with system toolchains, we're hitting new warnings during pretty much every platform enablement. I think defaulting to off is better for our build environments.

We could possibly change the default based on debug vs release mode, so builds on laptops still have Werror without needing to opt-in?

Reply:
I was thinking more inline with default to on, but we can set this to off in puppet-agent.

Reply:
That way we preserve the checks local development and Travis CI are doing.

Reply:
I'm going to go ahead and do what @MikaelSmith suggests since we need to get this in for Fedora 29.